### PR TITLE
Refactor LSP initialize method

### LINKED ISSUE #46: Refactor LSP initialize method
- [x] Refactor LSP _initialize_ method
- [x] _appveyor_ use python `3.5`
- [x]  _appveyor_ run flake8 and bandit checks
- [x] Fix python `3.5` compatibility
- [x] Add more tests

### PR DESCRIPTION
**Pull request by danixeee**
Closes #46.

Related to #48.

### CODE REVIEW

File: pygls/server.py [hunk 1]
```diff
@@ -105,7 +105,9 @@ class Server:
     """
 
     def __init__(self, protocol_cls, loop=None, max_workers=2):
-        assert issubclass(protocol_cls, asyncio.Protocol)
+        if not issubclass(protocol_cls, asyncio.Protocol):
+            raise TypeError(
```
Comment (augb) [inline pygls/server.py]: Extra newline unnecessary.

File: pygls/server.py [hunk 2]
```diff
@@ -212,7 +214,9 @@ class LanguageServer(Server):
     """
 
     def __init__(self, loop=None, protocol_cls=LanguageServerProtocol, max_workers: int = 2):
-        assert issubclass(protocol_cls, LanguageServerProtocol)
+        if not issubclass(protocol_cls, LanguageServerProtocol):
+            raise TypeError(
```
Comment (augb) [inline pygls/server.py]: Extra newline unnecessary.

File: pygls/uris.py [hunk 3]
```diff
@@ -29,6 +29,9 @@
 
 
 def _normalize_win_path(path):
+    if path is None:
+        return None
```
Comment (augb) [inline pygls/uris.py]: Should this return `None, None`?

Comment (danixeee) [reply pygls/uris.py]: Good catch. I will remove it completely and catch error where this function is called.

**Pull request merged**